From f4b0c45cc467d333beb130956d8ab1d5077a3910 Mon Sep 17 00:00:00 2001
From: Xu Xuehui <xxh@rock-chips.com>
Date: Thu, 4 Jan 2018 14:52:03 +0800
Subject: [PATCH] net: wireless: rockchip_wlan: fix ap6212a/ap6212 connect ap
 fail

The ap6212a/ap6212 firmware can not support MFP protocol

Change-Id: I9835e7da0e7dbac6554f75ff28c3c3e9e785f39d
Signed-off-by: Xu Xuehui <xxh@rock-chips.com>
---
 drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/wl_cfg80211.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/wl_cfg80211.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/wl_cfg80211.c
index e24608893f45..c935304d9e91 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/wl_cfg80211.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/wl_cfg80211.c
@@ -4820,7 +4820,7 @@ wl_cfg80211_set_mfp(struct bcm_cfg80211 *cfg,
 		mfp, wpa2_ie, rsn_cap[0], rsn_cap[1], fw_support));
 
 	if (fw_support == false) {
-		if (mfp) {
+		if (mfp == WL_MFP_REQUIRED) {
 			/* if mfp > 0, mfp capability set in wpa ie, but
 			 * FW indicated error for mfp. Propagate the error up.
 			 */
-- 
2.35.3

